sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar;
+
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.Functions;
+import io.crate.metadata.Scalar;
+import io.crate.operation.Input;
+import io.crate.operation.operator.input.ObjectInput;
+import io.crate.planner.symbol.Function;
+import io.crate.planner.symbol.Literal;
+import io.crate.planner.symbol.Symbol;
+import io.crate.testing.TestingHelpers;
+import io.crate.types.ArrayType;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.inject.ModulesBuilder;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class ArrayUniqueFunctionTest {
```

Comment:
Thank you for your feedback. I'll work on what is wrong.

2015-09-22 10:26 GMT-03:00 Matthias Wahl <notifications@github.com>:

> In
> sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java
> <https://github.com/crate/crate/pull/2619#discussion_r40084895>:
>
> > +import io.crate.types.DataTypes;
> > +import org.elasticsearch.common.inject.ModulesBuilder;
> > +import org.hamcrest.Matchers;
> > +import org.junit.Before;
> > +import org.junit.Rule;
> > +import org.junit.Test;
> > +import org.junit.rules.ExpectedException;
> > +
> > +import java.util.ArrayList;
> > +import java.util.Arrays;
> > +import java.util.List;
> > +
> > +import static org.hamcrest.Matchers.is;
> > +import static org.junit.Assert.assertThat;
> > +
> > +public class ArrayUniqueFunctionTest {
>
> Sorry for the many comments.
> These functions are really cool and valuable for crate, and it is great
> that you did them.
> Nonetheless, i had to comment on what is wrong.
> don't let this get you down, this is just the way we review code.
> in case you need it: http://instamotivation.com/ ;-)
>
> Keep up the good work!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crate/crate/pull/2619/files#r40084895>.
>



-- 
--
Glauber Machado Rodrigues